### ISSUE TITLE
Nav Bar || Dropdown React Component

### ISSUE THREAD

**Issue by vikasrohit** [appirio-tech/react-components#4]
This is a generic component which will be reused in many other components.

Component tree

- dropdown-menu
  - dropdown-menu-header
  - dropdown-menu-list

Requirements
- Create a branch off of master and make a PR to master
- Should transclude the content in dropdown-menu-header and dropdown-menu-list
- Should support abstracting the logic of opening and closing the dropdown list
- Should be configurable to show/hide menu pointer
  - Use `:after` pseudo-elements and rotate a rectangle that is clipped; we want a pure CSS pointer that also drops shadow.
- ~~Should support rendering the dropdown list into specified DOM element which would allow us to render the dropdown list in mobile view as slide down just below the nav header. (@vic-appirio do we need this feature to allow dropdowns to open in mobiles?)~~

**Comment by reynard2007** [comment]
Taking this.

**Comment by vikasrohit** [comment]
Go ahead. :+1:

**Comment by vikasrohit** [comment]
@reynard2007 Everything good so far?

**Comment by reynard2007** [comment]
Hi @vikasrohit, just had to travel without access to my machine, but I'm working on it.

**Comment by vikasrohit** [comment]
Cool. :+1:

**Comment by reynard2007** [comment]
Done with this, but where I'm traveling does not have reliable internet. Will push a merge request later tonight PHT.

**Comment by vikasrohit** [comment]
Ok. No problem. @reynard2007 Are you going to pick any other task today?

**Comment by vikasrohit** [comment]
@reynard2007 Please create PR against dev branch instead of master. We also have Panel component available there now.

**Comment by reynard2007** [comment]
Hi @vikasrohit, once I get back to the city, I'll pick #8.

**Comment by reynard2007** [comment]
It appears that #8 shoukd reuse this dropdiwn component. Should I carry on with it with what my implementation of the dropdown component even before review?

**Comment by vikasrohit** [comment]
@reynard2007 you can use that as base so that you don't get blocked. However, the PR for #8 should be done only after PR for this issue is closed and merged. So, please make the PR as soon as you can do. Once your branch is merged into dev, you can remove them from your branch for #8 and directly use the components from the dev branch.

**Comment by vikasrohit** [comment]
Just for information, we would not close the PR until the review and final fixes are done. So, you can continue fixing code in appropriate branches.

**Comment by reynard2007** [comment]
Now that I'm seeing the other requests, I will have to clarify the following after which I'll do a new PR (which should be quick):

1) The dropdown-header is the link you press that triggers the dropdown to open
2) The dropdown-menu-list will be inside the dropdown popup and will vary depending on whether it is used for quick links or user
3) We should have two transcluded content (as above) and these will change based on context, configurable with needed markup
4) Could you please expound on this: _Should support rendering the dropdown list into specified DOM element_

**Comment by vikasrohit** [comment]
Thanks @reynard2007 for working on this. I would try to answer your queries, let me know if that helps.

1. component should ask the className of the element under dropdown-header, which would be used as trigger. You can see the Panel component in the dev branch for this purpose.
2. Correct. In fact it can have any content, it depends on the parent component.
3. Yes, please refer Panel component which also do the multiple transclusion.
4. Basically, it means that when user clicks on the specified element under dropdown-header, the dropdown-menu-list should open at configurable place. In other words, if parent component has specified the place (via className or direct reference to the DOM element) to render the dropdown-menu-list, we should use that , otherwise, it should render itself as normal popup just below the dropdown-header. Our purpose of having this configuration, is to avoid popup and render the dropdown item inline in the DOM for mobile devices. You can hold on this requirement though, if it consumes time or is still confusing, for now. We can handle this in final fixes.

**Comment by vikasrohit** [comment]
@reynard2007 I am seeing two PR for the same issue. Please let me know which one is to be used for review, and please close the other one.

**Comment by vikasrohit** [comment]
@reynard2007 I am getting following error on other example pages.
`Uncaught TypeError: Cannot read property 'tagName' of null onClickOutside @ Dropdown.jsx:53`
It seems like the event handler is not being removed. Can you please verify if that is the problem with the component or is it used incorrectly in the example page?

**Comment by vikasrohit** [comment]
I have also update the requirements to not have the feature which was to allow rendering of the dropdown content in specified element.

**Comment by vikasrohit** [comment]
Closing task. Thanks @reynard2007 for your hard work.
